src/main/java/frc/robot/subsystems/climber/ClimberIOTalonFX.java

```diff
@@ -0,0 +1,225 @@
+    BaseStatusSignal.refreshAll(
+        leftVelocityRPSStatusSignal,
+        leftPositionRotationsStatusSignal,
+        leftStatorCurrentAmpsStatusSignal,
+        leftSetpointStatusSignal,
+        leftSupplyCurrentAmpsStatusSignal,
+        rightVelocityRPSStatusSignal,
+        rightPositionMetersStatusSignal,
+        rightStatorCurrentAmpsStatusSignal,
+        rightSetpointStatusSignal,
+        rightSupplyCurrentAmpsStatusSignal);
+
+    inputs.leftVelocityRPS = leftVelocityRPSStatusSignal.getValueAsDouble();
+    inputs.leftPositionMeters =
+        leftPositionRotationsStatusSignal.getValueAsDouble() * DRUM_CIRCUMFERENCE;
+    inputs.leftStatorCurrentAmps = leftStatorCurrentAmpsStatusSignal.getValueAsDouble();
+    inputs.leftSupplyCurrentAmps = leftSupplyCurrentAmpsStatusSignal.getValueAsDouble();
+    inputs.leftSetpoint = leftSetpointStatusSignal.getValueAsDouble();
+
+    inputs.rightVelocityRPS = rightVelocityRPSStatusSignal.getValueAsDouble();
+    inputs.rightPositionMeters = rightPositionMetersStatusSignal.getValueAsDouble();
+    inputs.rightStatorCurrentAmps = rightStatorCurrentAmpsStatusSignal.getValueAsDouble();
+    inputs.rightSupplyCurrentAmps = rightSupplyCurrentAmpsStatusSignal.getValueAsDouble();
+    inputs.rightSetpoint = rightSetpointStatusSignal.getValueAsDouble();
+
+    // inputs.closedLoopError = leftMotor.getClosedLoopError().getValue();
+    // inputs.power =
+    // inputs.controlMode = leftMotor.getControlMode().toString();
+
+    // update configuration if tunables have changed
+    if (kP.hasChanged()
+        || kI.hasChanged()
+        || kD.hasChanged()
+        || kS.hasChanged()
+        || kA.hasChanged()
+        || kV.hasChanged()
+        || kVExpo.hasChanged()
+        || kAExpo.hasChanged()) {
+      TalonFXConfiguration config = new TalonFXConfiguration();
+      MotionMagicConfigs motionMagicConfigs = new MotionMagicConfigs();
+
+      this.leftMotor.getConfigurator().refresh(config);
+      config.Slot0.kP = kP.get();
+      config.Slot0.kI = kI.get();
+      config.Slot0.kD = kD.get();
+      config.Slot0.kS = kS.get();
+      config.Slot0.kV = kV.get();
+      config.Slot0.kA = kA.get();
+      motionMagicConfigs.MotionMagicExpo_kV = kVExpo.get();
+      motionMagicConfigs.MotionMagicExpo_kA = kAExpo.get();
+
+      this.leftMotor.getConfigurator().apply(config);
+      this.rightMotor.getConfigurator().apply(config);
+    }
+  }
+
+  /**
+   * Set the leftMotor position to the specified value in degrees.
+   *
+   * @param position the position to set the leftMotor to in meters
+   * @param arbitraryFeedForward the arbitrary feed forward as a percentage of maximum power
+   */
+  @Override
+  public void setLeftMotorPosition(double position) {
+    leftMotor.setControl(
+        leftPositionCurrentRequest
+            .withPosition(Conversions.degreesToFalconRotations(position, GEAR_RATIO))); // FIXME: Feed forward should be acounted for already.
```

Comment:
We need to convert from meters of rope to rotations. The gear ratio is already accounted for based on the configuration done later in this class. Dividing the desired meters by the circumference of the drum, gives us the number of rotations.